packages/ensnode-utils/src/cache.ts

```diff
@@ -1,99 +1,101 @@
-import { PriorityQueue } from "./queue";
-
-export interface ICache<Key extends string, Value> {
+/**
+ * Cache that maps from string -> ValueType.
+ */
+export interface Cache<KeyType extends string, ValueType> {
   /**
-   * Store a value in the cache with the given key
+   * Store a value in the cache with the given key.
+   *
    * @param key Cache key
    * @param value Value to store
    */
-  set(key: Key, value: Value): void;
+  set(key: KeyType, value: ValueType): void;
 
   /**
-   * Retrieve a value from the cache
+   * Retrieve a value from the cache with the given key.
+   *
    * @param key Cache key
-   * @returns The stored value if exists and not expired, undefined otherwise
+   * @returns The cached value if it exists, otherwise undefined
    */
-  get(key: Key): Value | undefined;
+  get(key: KeyType): ValueType | undefined;
 
   /**
-   * Clean up resources and clear the cache
+   * Clear the cache.
    */
-  dispose(): void;
-}
-
-type Timestamp = number;
+  clear(): void;
```

Comment:
Nice update, now I see why I didn't like the `dispose` method on my `ICache` interface.